### DISCUSSION TITLE
Install Plugin Cucumber

### DISCUSSION THREAD

**Discussion by rohitmit07** [eclipse-theia/theia#8664] [Q&A] [answered]
How to install Plugin Cucumber in Theia

Plugin is available for both Eclipse & VS  

https://github.com/cucumber/cucumber-eclipse
https://marketplace.visualstudio.com/items?itemName=alexkrechik.cucumberautocomplete

**Comment by vince-fugnitto** [comment, marked as answer]
@rohitmit07 thank you for the question, it ultimately depends on your use-case:

1. if you'd like to simply add the plugin for yourself then you can simply download and save the extension under the `plugins` folder which is referenced when starting the application:

https://github.com/eclipse-theia/theia/blob/ecfe920698e4eea5afd379ca361b263c3fc970a0/examples/browser/package.json#L61

2. if you'd like the plugin to be available to your users through the `extensions` view then a version will need to be added to the open-source marketplace [open-vsx](https://github.com/eclipse/openvsx) so it can be fetched and installed. If it is not available in the _open-vsx_ then you can either request to add it or publish it yourself.

3. if you'd like the plugin to be available by **default** for your application (`builtin`) then you need to make sure that step 2 is completed, and the plugin is referenced in your application's `package.json` like so:

https://github.com/eclipse-theia/theia/blob/ecfe920698e4eea5afd379ca361b263c3fc970a0/package.json#L85-L87

When using the script `yarn download:plugins` it will fetch and download all plugins (by their URL) and place them under a `plugins` folder which is referenced when the application is started.
